polkadot/xcm/src/v3/traits.rs

```diff
@@ -143,6 +143,9 @@ pub enum Error {
 	/// An asset cannot be deposited, probably because (too much of) it already exists.
 	#[codec(index = 34)]
 	NotDepositable,
+	/// Too many assets matched the given asset filter.
+	#[codec(index = 35)]
+	TooManyAssets,
 
 	// Errors that happen prior to instructions being executed. These fall outside of the XCM
```

Comment:
This is kind of strange or unclear that we are just changing the encoding for the errors below, I can see:
```
// Errors that happen prior to instructions being executed. These fall outside of the XCM
// spec.
```
Those bellow errors are not part of the spec, so changing the encoding might be okay, could be. But also, for example, I think this no longer applies to `ExceedsStackLimit`, which is used with the [`FrameTransactionalProcessor`](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-builder/src/transactional.rs#L38) during instruction execution.

I'm not sure what the best way to fix or clean this up is, but we should clarify this somehow with a follow-up. Maybe we could add an index to the `ExceedsStackLimit` or remove those errors from here - I'm not sure :)

Reply:
I agree `ExceedsStackLimit` should be revisited given we use it in instructions.
I think this new error should be part of a v5 error.